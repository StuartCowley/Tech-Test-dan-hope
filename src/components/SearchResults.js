import React from "react";
import "../styles/searchResults.css";
import PropTypes from "prop-types";

// eslint-disable-next-line react/function-component-definition
const SearchResults = ({ results }) => {
  if (!results.length) {
    return <p className="para-fail">No results</p>;
    // eslint-disable-next-line no-else-return
  } else {
    return (
      <>
        {results.map((image) => (
          <>
            <div className="all-images" />
            <img className="space-image" src={image} alt="mock" key={image} />
          </>
        ))}
      </>
    );
  }
};

export default SearchResults;

SearchResults.propTypes = {
  // eslint-disable-next-line react/require-default-props
  results: PropTypes.arrayOf(PropTypes.string),
};
