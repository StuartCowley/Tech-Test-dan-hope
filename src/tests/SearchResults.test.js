import React from "react";
import { render } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("SearchResults", () => {
  const validProps = {
    results: ["i1", "i2, i3"],
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchResults results={validProps.results} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
