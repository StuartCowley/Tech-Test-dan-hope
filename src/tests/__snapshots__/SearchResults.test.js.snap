// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly 1`] = `
<DocumentFragment>
  <p>
    Search Results
  </p>
  <img
    alt="mock"
    class="card-image"
    src="i1"
  />
  <p>
    Search Results
  </p>
  <img
    alt="mock"
    class="card-image"
    src="i2, i3"
  />
</DocumentFragment>
`;
